Unable to build Quarkus main - Gradle plugin - can't find plexus-utils-3.4.1.jar

### Describe the bug

Unable to build Quarkus main - Gradle plugin - can't find plexus-utils-3.4.1.jar

This block our daily and PR CI runs as we can't build Quarkus main - e.g. https://github.com/quarkus-qe/quarkus-test-suite/runs/7029470577?check_suite_focus=true

I was able to reproduce the same locally:
 - using custom local repo
 - also when I removed `~/.m2/repository/org/codehaus/plexus/plexus-utils`

Error:
```
[INFO] Building Quarkus - Gradle Plugin 999-SNAPSHOT                 [803/1073]
...
[INFO] --- exec-maven-plugin:3.0.0:exec (gradle) @ io.quarkus.gradle.plugin ---
To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.4.2/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build
> Task :gradle-application-plugin:clean UP-TO-DATE
> Task :gradle-model:compileJava UP-TO-DATE
> Task :gradle-application-plugin:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':gradle-application-plugin:compileJava'.
> Could not resolve all files for configuration ':gradle-application-plugin:compileClasspath'.
   > Could not find plexus-utils-3.4.1.jar (org.codehaus.plexus:plexus-utils:3.4.1).
     Searched in the following locations:
         file:/Users/rsvoboda/Downloads/quarkus-main-local-repo/org/codehaus/plexus/plexus-utils/3.4.1/plexus-utils-3.4.1.jar
```

Part of stacktrace
```
* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':gradle-application-plugin:compileJava'.
	at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:38)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.executeTask(EventFiringTaskExecuter.java:77)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:55)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter$1.call(EventFiringTaskExecuter.java:52)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:204)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$CallableBuildOperationWorker.execute(DefaultBuildOperationRunner.java:199)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.call(DefaultBuildOperationRunner.java:53)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:73)
	at org.gradle.api.internal.tasks.execution.EventFiringTaskExecuter.execute(EventFiringTaskExecuter.java:52)
	at org.gradle.execution.plan.LocalTaskNodeExecutor.execute(LocalTaskNodeExecutor.java:74)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionGraph$InvokeNodeExecutorsAction.execute(DefaultTaskExecutionGraph.java:333)
...
Caused by: org.gradle.internal.resolve.ArtifactNotFoundException: Could not find plexus-utils-3.4.1.jar (org.codehaus.plexus:plexus-utils:3.4.1).
Searched in the following locations:
    file:/Users/rsvoboda/Downloads/quarkus-main-local-repo/org/codehaus/plexus/plexus-utils/3.4.1/plexus-utils-3.4.1.jar
	at org.gradle.internal.resolve.result.DefaultBuildableArtifactResolveResult.notFound(DefaultBuildableArtifactResolveResult.java:28)
	at org.gradle.api.internal.artifacts.repositories.resolver.ExternalResourceResolver$RemoteRepositoryAccess.resolveArtifact(ExternalResourceResolver.java:509)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.CachingModuleComponentRepository$ResolveAndCacheRepositoryAccess.resolveArtifact(CachingModuleComponentRepository.java:469)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.LocalModuleComponentRepository$LocalAccess.resolveArtifact(LocalModuleComponentRepository.java:101)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ErrorHandlingModuleComponentRepository$ErrorHandlingModuleComponentRepositoryAccess.lambda$resolveArtifact$15(ErrorHandlingModuleComponentRepository.java:181)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ErrorHandlingModuleComponentRepository$ErrorHandlingModuleComponentRepositoryAccess.tryResolveAndMaybeBlacklist(ErrorHandlingModuleComponentRepository.java:237)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ErrorHandlingModuleComponentRepository$ErrorHandlingModuleComponentRepositoryAccess.performOperationWithRetries(ErrorHandlingModuleComponentRepository.java:203)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.ErrorHandlingModuleComponentRepository$ErrorHandlingModuleComponentRepositoryAccess.resolveArtifact(ErrorHandlingModuleComponentRepository.java:179)
	at org.gradle.api.internal.artifacts.ivyservice.ivyresolve.RepositoryChainArtifactResolver.resolveArtifact(RepositoryChainArtifactResolver.java:100)
	at org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.DefaultArtifactSet$LazyArtifactSupplier.calculateValue(DefaultArtifactSet.java:284)
	at org.gradle.api.internal.artifacts.ivyservice.resolveengine.artifact.DefaultArtifactSet$LazyArtifactSupplier.calculateValue(DefaultArtifactSet.java:255)
	at org.gradle.internal.model.CalculatedValueContainer$CalculationState.lambda$attachValue$0(CalculatedValueContainer.java:218)
	at org.gradle.internal.Try.ofFailable(Try.java:41)
```

### Expected behavior

Build passes

### Actual behavior

Build fails

### How to Reproduce?

git clone https://github.com/quarkusio/quarkus.git && cd quarkus && ./mvnw -B -s .github/mvn-settings.xml -Dmaven.repo.local=${HOME}/Downloads/quarkus-main-local-repo clean install -Dquickly -Prelocations


### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

Java 11 / Java 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

Quarkus main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/devtools/gradle/build.gradle
+++ b/devtools/gradle/build.gradle
@@ -21,14 +21,7 @@ subprojects {
     }
 
     repositories {
-        // in case a custom local repo is configured we are going to use that instead of the default mavenLocal()
-        if (System.properties.containsKey('maven.repo.local')) {
-            maven {
-                url System.properties.get('maven.repo.local')
-            }
-        } else {
-            mavenLocal()
-        }
+        mavenLocal()
         mavenCentral()
     }
 

